src/core/include/mp-units/numbers.h

```diff
@@ -0,0 +1,294 @@
+template<typename T, typename U>
+concept compound_scales_with = scales_with<T, U> && detail::compound_multiplication_with<T, U>;
+
+}  // namespace detail
+
+template<typename T, typename U>
+concept scalar_for = detail::scales_with<U, T>;
+template<typename T, typename U>
+concept field_for = scalar_for<T, U> && detail::division_with<U, T>;
+template<typename T, typename U>
+concept compound_scalar_for = detail::compound_scales_with<U, T>;
+template<typename T, typename U>
+concept compound_field_for = compound_scalar_for<T, U> && detail::compound_division_with<U, T>;
+
+template<typename T>
+concept vector_space = point_space<T> && detail::compound_scales_with<T, number_scalar_t<T>>;
+template<typename T>
+concept f_vector_space = vector_space<T> && detail::compound_division_with<T, number_scalar_t<T>>;
+
+template<typename T>
+concept field_number = f_vector_space<T> && detail::compound_scales_with<T, T>;
+
+template<typename T>
+concept field_number_line = field_number<T> && number_line<T>;
+
+template<typename T>
+concept scalar_number = field_number<T> && (field_number_line<T> || is_complex_number_v<T>);
+
+template<typename T>
+struct is_number<const T> : is_number<T> {};
+template<typename T, typename U>
+struct is_number<std::chrono::time_point<T, U>> : std::true_type {};
+#if __cpp_lib_chrono >= 201803
+template<>
+struct is_number<std::chrono::day> : std::true_type {};
+template<>
+struct is_number<std::chrono::month> : std::true_type {};
+template<>
+struct is_number<std::chrono::year> : std::true_type {};
+template<>
+struct is_number<std::chrono::weekday> : std::true_type {};
+template<>
+struct is_number<std::chrono::year_month> : std::true_type {};
+#endif
+
+template<typename T>
+struct is_complex_number<const T> : is_complex_number<T> {};
+template<typename T>
+struct is_complex_number<std::complex<T>> : std::true_type {};
+
+namespace detail {
+
+template<typename T>
+concept inferable_identities = std::common_with<T, number_difference_t<T>> && std::constructible_from<T, int>;
+
+template<detail::inferable_identities T>
+struct inferred_number_zero<T> {
+  auto static constexpr value = T(0);
+};
+template<detail::inferable_identities T>
+struct inferred_number_one<T> {
+  auto static constexpr value = T(1);
+};
+
+}  // namespace detail
+
+template<typename T>
+struct number_zero<const T> : number_zero<T> {};
+template<typename T>
+struct number_one<const T> : number_one<T> {};
+
+template<typename T>
+struct number_scalar<const T> : number_scalar<T> {};
+template<std::integral T>
+struct number_scalar<T> : std::type_identity<T> {};
+template<std::floating_point T>
+struct number_scalar<T> : std::type_identity<T> {};
+template<typename T>
+struct number_scalar<std::complex<T>> : std::type_identity<T> {};
+template<typename T, typename U>
+struct number_scalar<std::chrono::duration<T, U>> : std::type_identity<T> {};
```

Comment:
Should there also be a specialization for `time_point` as it is explicitly specialized to be a number?